pkg/controller/monitor/monitor_controller.go

```diff
@@ -415,6 +456,37 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 	return reconcile.Result{}, nil
 }
 
+func fillDefaults(instance *operatorv1.Monitor) {
+	if instance.Spec.ExternalPrometheus != nil && instance.Spec.ExternalPrometheus.ServiceMonitor != nil {
```

Comment:
Is it ever valid to have `Spec.ExternalPrometheus != nil` and `Spec.ExternalPrometheus.ServiceMonitor == nil` ?

Do we need a validation function for this resource as well? I think probably yes

Reply:
A user may want to get the TLS bundle without Tigera creating the Service monitor. In that case ExternalPrometheus is not nil.

Since we no longer embed all of the prometheus ServiceMonitor any longer, but only a subset we no longer have any need for validation, so I removed it. Fields related to being able to access our Prometheus are controlled and defaulted by us. I don't think there are any contradictory configurations possible any longer.